source/common/http/http1/conn_pool.cc

```diff
@@ -81,6 +81,7 @@ ConnectionPool::Cancellable* ConnPoolImpl::newStream(StreamDecoder& response_dec
     ready_clients_.front()->moveBetweenLists(ready_clients_, busy_clients_);
     ENVOY_CONN_LOG(debug, "using existing connection", *busy_clients_.front()->codec_client_);
     attachRequestToClient(*busy_clients_.front(), response_decoder, callbacks);
+    busy_clients_.front()->disableIdleTimer();
```

Comment:
I'm not familiar enough with this code to be sure, but is there a potential race here where the idle timer can expire after the request is attached, but before the timer is disabled? What will happen in that case?

Reply:
That is a good point. Should I move the disableTimer before attachRequestToClient? Would it help?